### DISCUSSION TITLE
Automatically update etc/hosts

### DISCUSSION THREAD

**Discussion by diegovogel** [laravel/valet#1401] [Ideas]
Is Valet supposed to automatically update the `etc/hosts` file? I've heard that maybe it used to, but it doesn't anymore. There's no mention of the hosts file in the docs so it's unclear if this is a bug or a missing feature.

Anyway every time I use `valet link` I have to open up `etc/hosts` and add the new domain there. It would be great if Valet did this automatically.

**Comment by drbyte** [comment]
No, Valet doesn't touch the `etc/hosts` file. It doesn't need to.

Perhaps you have a custom DNS entry in your MacOS network configuration, ahead of 127.0.0.1, so that dnsmasq isn't getting called?

**Reply by diegovogel** [reply]
Ok thanks, good to know I shouldn't have to update the hosts file. I'm not sure what you mean by custom DNS entry. How would I check that?

**Reply by drbyte** [reply]
System Preferences->Network->Advanced->DNS
<img width="642" alt="dns127-0-0-1" src="https://user-images.githubusercontent.com/404472/232182930-a8fb5bc5-3a72-45c7-a295-1063614fbf33.png">

**Reply by diegovogel** [reply]
Oh gotcha. I set up a resolver to only send my local TLD (.off) to dnsmasq [as outlined in option #2 here](https://www.stevenrombauts.be/2018/01/use-dnsmasq-instead-of-etc-hosts/#2-only-send-test-and-box-queries-to-dnsmasq).

I think this may actually be a Chrome issue. If I ping any `.off` domain it resolves to 127.0.0.1 as expected.

![image](https://user-images.githubusercontent.com/23545422/232184304-0b0b600a-c2ae-4ca2-8cc0-a3b5044dc1bd.png)

And I just tested in Firefox and Safari and had no problem viewing my app. So only Chrome can't resolve the domain.

![image](https://user-images.githubusercontent.com/23545422/232184338-950ebbfb-ff4b-4e35-bab7-3683ccaf5540.png)

I've already tried clearing Chrome's DNS cache. Any idea what else I can try?

**Reply by diegovogel** [reply]
Finally got it to work! In case anyone else needs a solution: it was because of Chrome's secure DNS setting (`chrome://settings/security`). I switched it to "With your current service provide", but disabling it entirely would probably work too.

![image](https://user-images.githubusercontent.com/23545422/232185269-9b58fa38-109a-479b-95f8-b412f75dd6d5.png)

Other things I did along the way that may have helped:
- Cleared DNS cache: `chrome://net-internals/#dns`
- Flushed socket pools: `chrome://net-internals/#sockets`
- Cleared browsing data for all time: `chrome://settings/clearBrowserData`

![image](https://user-images.githubusercontent.com/23545422/232185422-75756292-8f03-4e1b-91cb-7f5530ae6904.png)

@drbyte thanks for your help!
